src/Microsoft.CodeAnalysis.Analyzers/Core/AnalyzerBannedSymbols.txt

```diff
@@ -0,0 +1,10 @@
+T:System.IO.File; Do not do file IO in analyzers
+T:System.IO.Directory; Do not do file IO in analyzers
+M:System.IO.Path.GetTempPath; Do not do file IO in analyzers
+T:System.Environment; Analyzers should not read their settings directly from environment variables
+M:System.Reflection.Assembly.Load(System.Byte[]); Analyzers should only load their dependencies via standard runtime mechanisms
+M:System.Reflection.Assembly.Load(System.String); Analyzers should only load their dependencies via standard runtime mechanisms
+M:System.Reflection.Assembly.Load(System.Reflection.AssemblyName); Analyzers should only load their dependencies via standard runtime mechanisms
+M:System.Reflection.Assembly.Load(System.Byte[],System.Byte[]); Analyzers should only load their dependencies via standard runtime mechanisms
+P:System.Globalization.CultureInfo.CurrentCulture; Analyzers should use the locale given by the compiler command line arguments, not the CurrentCulture
+P:System.Globalization.CultureInfo.CurrentUICulture; Analyzers should use the locale given by the compiler command line arguments, not the CurrentUICulture
```

Comment:
@RikkiGibson Should all "analyzers" in these messages be updated to "analyzers or generators"